### PR TITLE
Drop more deprecated APIs

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , this is the last (!) batch of straightforward API drops

File: tools/crm_mon.c [hunk 1]
```diff
@@ -722,16 +722,9 @@ static GOptionEntry display_entries[] = {
...
-     * writing:
-     * * galera
-     * * pgsql
-     * * rabbitmq-cluster
-     * * redis
-     *
-     * Removing this option must wait until those are updated and we no longer
-     * support building on any platforms that ship the older agents in stock
-     * packages.
+    /* @COMPAT resource-agents <4.15.0 uses --as-xml, so removing this option
```
Comment (nrwahl2) [inline tools/crm_mon.c:725]: This reads as though `resource-agents` 4.15.0 no longer uses `--as-xml`, which is not the case. The main branch still does.

**Review by nrwahl2 — COMMENTED** [review]
Nice!

File: tools/crm_mon.c (hunk 1 above)
Comment (kgaillot) [reply tools/crm_mon.c:725]: I thought that was fixed in 4.15.0

Comment (kgaillot) [reply tools/crm_mon.c:725]: never mind, it did get fixed: it chooses the correct argument based on the pacemaker feature set

Comment (nrwahl2) [reply tools/crm_mon.c:725]: ah yep, my bad

**Pull request merged**